### PR TITLE
make caFile option configurable

### PR DESCRIPTION
**Pull request by vincentvilo-aws**
*Description of changes:* Making caFile option configurable in KPL configuration 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws/kinesis/core/kinesis_producer.cc [hunk 1]
```diff
@@ -137,6 +139,7 @@ make_sdk_client_cfg(const aws::kinesis::core::Configuration& kpl_cfg,
   cfg.executor = sdk_client_executor;
   cfg.verifySSL = kpl_cfg.verify_certificate();
   cfg.caPath = ca_path;
+  cfg.caFile = ca_file;
```
Comment (akidambisrinivasan) [inline aws/kinesis/core/kinesis_producer.cc:142]: Lets only set this when its not empty

File: aws/kinesis/main.cc [hunk 2]
```diff
@@ -150,6 +153,9 @@ void process_options(int argc, char* const* argv) {
     case 'a':
         options.ca_path = std::string(optarg);
         break;
+    case 'f':
+        options.ca_file = std::string(optarg);
```
Comment (akidambisrinivasan) [inline aws/kinesis/main.cc:157]: We dont need this, java is passing it as env variable

File: aws/kinesis/main.cc [hunk 3]
```diff
@@ -64,6 +65,7 @@ struct option long_opts[]{
         {"log-level",              required_argument, NULL, 'l'},
         {"enable-stack-trace",     no_argument,       NULL, 't'},
         {"ca-path",                required_argument, NULL, 'a'},
+        {"ca-file",                required_argument, NULL, 'f'},
```
Comment (akidambisrinivasan) [inline aws/kinesis/main.cc:68]: We dont need this

File: aws/kinesis/main.cc [hunk 4]
```diff
@@ -118,6 +120,7 @@ void usage(const std::string program_name, const std::string& message) {
     option_description("-l", "--log-level", "Controls the level of detail emitted from the producer.  Valid: ['trace', 'debug', 'info', 'warn', 'error', 'fatal']");
     option_description("-t", "--enable-stack-trace", "Will dump a stack trace and abort if certain memory errors are triggered");
     option_description("-a", "--ca-path", "Location of the CA root certificate that the producer will use for TLS connections.");
+    option_description("-f", "--ca-file", "File of the CA root certificate that the producer will use for TLS connections.");
```
Comment (akidambisrinivasan) [inline aws/kinesis/main.cc:123]: We dont need this either, I am not aware of besides KPL java who runs the c++ code directly. So we can skip this for now and add it later if its needed

File: aws/kinesis/main.cc [hunk 5]
```diff
@@ -349,6 +355,23 @@ std::string get_ca_path() {
   return p;
 }
 
+
+std::string get_ca_file() {
+  std::string f = "";
+  if (!options.ca_file.empty()) {
```
Comment (akidambisrinivasan) [inline aws/kinesis/main.cc:361]: Dont need this either

File: aws/kinesis/main.cc [hunk 6]
```diff
@@ -349,6 +355,23 @@ std::string get_ca_path() {
...
+    f = options.ca_file;
+  } else {
+    auto v = std::getenv("CA_FILE");
+
+    if (v) {
+      f = v;
+    }
+  }
+  LOG(info) << "Setting CA file to " << f;
```
Comment (akidambisrinivasan) [inline aws/kinesis/main.cc:370]: Lets only print this if its not an empty string

File: aws/kinesis/core/kinesis_producer.cc (hunk 1 above)
Comment (SergeyRyabinin) [reply aws/kinesis/core/kinesis_producer.cc:142]: > Lets only set this when its not empty

In the [AWS C++ SDK](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/include/aws/core/client/ClientConfiguration.h#L290), it is defined as 
> ```Aws::String caFile;```

where Aws::String is aliased to std::string. Setting an empty string would take no effect, so there is no need to check if ca_file is empty or not.

**Review by SergeyRyabinin — APPROVED** [review]
looks good from AWS SDK C++ point of view.

File: aws/kinesis/core/kinesis_producer.cc (hunk 1 above)
Comment (vincentvilo-aws) [reply aws/kinesis/core/kinesis_producer.cc:142]: ack, resolving this thread

**Review by akidambisrinivasan — APPROVED** [review]

File: aws/kinesis/main.cc [hunk 7]
```diff
@@ -389,7 +404,8 @@ int main(int argc, char* const* argv) {
           creds_providers.first,
           creds_providers.second,
           executor,
-          ca_path);
+          ca_path,
+          ca_file);
```
Comment (akidambisrinivasan) [inline aws/kinesis/main.cc:408]: How come this doesnt need a change in kinesis_producer.h ? Does this compile ?

**Pull request merged**